### PR TITLE
Refactor cluster roles for minimal rbac access

### LINKED ISSUE #1389: Improve Rook's Security Model
Rook currently uses multiple namespaces and service accounts. The rook operator can create and delete namespaces, service accounts, roles and role bindings on the fly. As a result rook requires a *very* privileged service account (see [here](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/rook-operator.yaml#L9)).

For production clusters this not going to work. We need to explore running in a mode where we do not have privileges to destroy the entire cluster. Also creating a deleting namespace is not necessarily something that every cluster-admin is going to want us to do.

If a cluster-admin wants to set policies (PSP, SCC, resource quotas, etc.) they need to be able to control which service accounts and namespaces Rook uses. Creating namespaces and service accounts on the fly means that these policies would need to be dynamically applied too.

This is a complicated issue but it's one that we should front load. Its come up in the following places #1314 #1337 #1348 #1211 and possible more. Also discussion on twitter here: https://twitter.com/the_sttts/status/945077544587616257

 I'm going to write a design doc and link it here when its ready.

### PR DESCRIPTION
**Pull request by travisn**
Signed-off-by: travisn <tnielsen@redhat.com>

Description of your changes:
Per the [security model](https://github.com/rook/rook/blob/master/design/security-model.md#proposed-changes) proposal:
- There is one serviceAccount per namespace, one for `rook-system` and one for each cluster that is created: `rook-ceph-cluster`.  
- The operator, agent, and discovery daemons all run with the `rook-system` service account
- The OSD daemon runs with the `rook-ceph-cluster` service account. Other cluster daemons (mon, mgr, mds, rgw) run with the default service account since they do not require access to k8s resources.
- Rook no longer creates service accounts, cluster roles, roles, etc. These must be created by the admin.

Differing from the security model spec, a ClusterRoleBinding was needed for several resources:
- Rook CRDs: Watched and handled across all namespaces
- Pods: The agent needs to analyze pods for fencing
- PVs & PVCs: The operator provisioner needs to create, delete, list, etc
- Nodes: Used to determine on which nodes to place mons
- Storage classes: watched by the operator

The `rook-ceph-core` role was created for the namespace-specific resources and the `rook-ceph-global` role was created for cluster-wide resources.

Which issue is resolved by this Pull Request:
Resolves #1389

Checklist:
- [x] Documentation has been updated, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Test upgrade from v0.7.1

### CODE REVIEW

**Comment by bassam** [comment]
I have to take a deeper look but it still looks like we are granting the rook operator many unnecessary privileges: https://github.com/travisn/rook/blob/aa335283f0b3c0b4cbfda7509117d4d9c1365e23/cluster/examples/kubernetes/ceph/operator.yaml#L98

**Comment by travisn** [comment]
The `rook-ceph-core` are the privileges that need to be granted on a namespace-specific basis for managing a cluster. The operator needs access to create secrets, pods, services, etc in a specific namespace. 

One thing that could be restricted more is that all of these `rook-ceph-core` privileges are not necessary in the `rook-ceph-system` namespace. For example, the operator does not need to create secrets in the system namespace, only in a cluster namespace. By making these privileges more fine-grained, I see that we would need to define three cluster roles:
- System namespace management: Privileges to start the agent and discovery daemons
- Cluster namespace management: Privileges to start a ceph cluster in a specific namespace, including ceph daemon pods, secrets, services, etc.
- Kubernetes-wide management: Privileges to watch resources that are not namespace-specific or could be in an arbitrary namespace, such as watching nodes, storage classes, and watching pods for fencing.

**Comment by travisn** [comment]
@bassam before I make the change for my last suggestion do you have any other feedback?

**Comment by travisn** [comment]
@bassam shall i proceed with my last recommendation or any other feedback?

File: Documentation/upgrade.md [hunk 1]
```diff
@@ -145,6 +145,64 @@ kubectl delete clusterrolebindings rook-agent
...
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: rook-ceph-cluster
+subjects:
+- kind: ServiceAccount
+  name: rook-ceph-cluster
+  namespace: rook
+```
+
+Now create the yaml:
```
Comment (dmick) [inline Documentation/upgrade.md]: "create the objects in the cluster"

**Comment by travisn** [comment]
@bassam I went ahead and further restrict the system namespace with the `rook-ceph-system` Role. Your review would be appreciated!

**Pull request merged**